internal/messaging/response_handler.go

```diff
@@ -1,147 +1,536 @@
+	buyer := common.HexToAddress(request.MintRequest.BuyerAddress)
+
+	h.logger.Debugf("abi: %v", abi)
+
+	// TODO @evlekht unhardocoded, figure out what it is at all
+	// uri := "data:application/json;base64,eyJuYW1lIjoiQ2FtaW5vIE1lc3NlbmdlciBCb29raW5nVG9rZW4gVGVzdCJ9Cg=="
+
+	// Get a Token URI for the token.
+	jsonPlain, uri, err := createTokenURIforMintResponse(response.MintResponse)
+	if err != nil {
+		addErrorToResponseHeader(response, fmt.Sprintf("error creating token URI: %v", err))
+		return true
+	}
+
+	h.logger.Debugf("Token URI JSON: %s\n", jsonPlain)
+
+	if response.MintResponse.BuyableUntil == nil || response.MintResponse.BuyableUntil.Seconds == 0 {
+		response.MintResponse.BuyableUntil = timestamppb.New(time.Now().Add(time.Second * time.Duration(h.cfg.BuyableUntilDefault)))
+	}
+	// MINT TOKEN
+	txID, tokenID, err := mint(
+		h.ethClient,
+		bookingTokenAddress,
+		abi,
+		h.pk.ToECDSA(),
+		buyer,
+		uri,
+		big.NewInt(response.MintResponse.BuyableUntil.Seconds),
+	)
 	if err != nil {
 		errMessage := fmt.Sprintf("error minting NFT: %v", err)
 		if errors.Is(err, context.DeadlineExceeded) {
-			errMessage = fmt.Sprintf("%v: %v", tvm.ErrAwaitTxConfirmationTimeout, h.tvmClient.Timeout)
+			errMessage = fmt.Sprintf("transaction deadline exceeded: %v", evm.ErrAwaitTxConfirmationTimeout)
 		}
 		addErrorToResponseHeader(response, errMessage)
 		return true
 	}
-	if !success {
-		addErrorToResponseHeader(response, "minting NFT tx failed")
-		return true
-	}
+
 	h.logger.Infof("NFT minted with txID: %s\n", txID)
 	response.MintResponse.Header.Status = typesv1alpha.StatusType_STATUS_TYPE_SUCCESS
-	response.MintTransactionId = txID.String()
+	response.MintResponse.BookingToken = &typesv1alpha.BookingToken{TokenId: int32(tokenID.Int64())}
+	response.MintTransactionId = txID
 	return false
 }
 
-func (h *TvmResponseHandler) handleMintRequest(ctx context.Context, response *ResponseContent) bool {
+func (h *EvmResponseHandler) handleMintRequest(_ context.Context, response *ResponseContent) bool {
 	if response.MintResponse.Header == nil {
 		response.MintResponse.Header = &typesv1alpha.ResponseHeader{}
 	}
 	if response.MintTransactionId == "" {
 		addErrorToResponseHeader(response, "missing mint transaction id")
 		return true
 	}
-	mintID, err := ids.FromString(response.MintTransactionId)
+
+	abi, err := loadABI(h.cfg.BookingTokenABIFile)
 	if err != nil {
-		addErrorToResponseHeader(response, fmt.Sprintf("error parsing mint transaction id: %v", err))
+		addErrorToResponseHeader(response, fmt.Sprintf("error loading ABI: %v", err))
 		return true
 	}
 
-	success, txID, err := h.tvmClient.SendTxAndWait(ctx, transferNFTAction(h.tvmClient.Address(), mintID))
+	value64 := uint64(response.BookingToken.TokenId)
+	tokenID := new(big.Int).SetUint64(value64)
+
+	bookingTokenAddress := common.HexToAddress("0xd4e2D76E656b5060F6f43317E8d89ea81eb5fF8D")
```

Comment:
Didn't we want to unhardcode it?
@havan wdyt?

Reply:
We already did that in the `handleMintResponse` function. @VjeraTurk we need to use the config variable in this function also.